cpp/CMakeLists.txt

```diff
@@ -223,6 +223,8 @@ target_link_libraries(
 )
 
 target_compile_features(raft INTERFACE cxx_std_17 $<BUILD_INTERFACE:cuda_std_17>)
+target_compile_options(raft PUBLIC $<$<COMPILE_LANGUAGE:CUDA>:--expt-extended-lambda --expt-relaxed-constexpr>)
```

Comment:
```suggestion
target_compile_options(raft PUBLIC $<$<COMPILE_LANGUAGE_AND_ID:CUDA,NVIDIA>:--expt-extended-lambda --expt-relaxed-constexpr>)
```

Reply:
Since this is public we should be extra safe and only apply this to `nvcc` and not clang ( for the edgiest of edge cases )

Reply:
Thanks Robert! Just committed that change